Obsidian/Net/Packets/Play/Serverbound/PlayerBlockPlacement.cs

```diff
@@ -54,8 +56,9 @@ public async ValueTask HandleAsync(Server server, Player player)
                     Server.LastInventoryId = 1;
 
                 var maxId = Math.Max((byte)1, ++Server.LastInventoryId);
+                var bm = (BlockMeta)server.World.GetBlockMeta(position);
 
-                if (server.World.GetBlockMeta(position) is BlockMeta meta && meta.InventoryId != Guid.Empty)
+                if (bm is BlockMeta meta && meta.InventoryId != Guid.Empty)
```

Comment:
Unnecessary casting

Reply:
GetBlockMeta returns a nullable type

Reply:
If you're checking the type in this if statement while also outputting a variable then it doesn't matter if it's nullable because it will be false if it's null.

```cs
                var bm = server.World.GetBlockMeta(position);
                // This won't be run if bm is null, and if it isn't null, meta is defined and not null.
                if (bm is BlockMeta meta && meta.InventoryId != Guid.Empty)
```

Reply:
FYI, the `is (type)` is doing type casting for you, `meta` is implictly set to the type you are checking against.

https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/operators/is

Reply:
Ahh, got it. I hadn't actually read through the code past fixing the build error :)